server/src/main/java/org/elasticsearch/cluster/NotMasterException.java

```diff
@@ -14,7 +14,13 @@
 import java.io.IOException;
 
 /**
- * Exception which indicates that an operation failed because the node stopped being the elected master.
+ * Exception indicating that an operation failed because the node stopped being the elected master.
+ * Since this exception is thrown prior to the cluster state publication, it should only be used when the cluster state update
+ * *definitely* did not happen, and there is no possibility the next master committed the cluster state update.
+ *
+ * This is different to the {@code FailedToCommitClusterStateException}
```

Comment:
Suggest using `@link` so that it links to the other class. Also grammar/punctionation nits :)

```suggestion
 * This is different from {@link FailedToCommitClusterStateException}.
```